src/msg.c

```diff
@@ -262,6 +263,17 @@ static bool session_node_fd_cmp(const void *entry_data, const void *user_data)
 	return session->node_fd == node_fd;
 }
 
+static bool session_id_cmp(const void *entry_data, const void *user_data)
+{
+	const struct session *session = entry_data;
+	char id[17]; /* Use KNOT_ID_LEN */
+	const char *device_id = user_data;
+
+	snprintf(id, sizeof(id), "%016"PRIx64, session->id);
```

Comment:
Why `%016`?

Reply:
Formatting for print 16 characters from inttypes.h

Reply:
But why 16?

Reply:
`%016` will put 0's in the string depending which is the value of `session->id`
if `session->id` is equal to `10`  the string `id` will `000000...a` in hexadecimal value, which represents the macro `PRIx64` (**pri**nt **i**nteger 64 bits in he**x**decimal)

Reply:
Also, the maximum that a `uint_64` can fit is 8 bytes which in hexadecimal represents 16 characters.